### PR TITLE
Migrating SCSt Schema Registry Sample

### PR DESCRIPTION
**Pull request by sobychacko**

### CODE REVIEW

File: tools/kafka/docker-compose/mysql.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: '3'
```
Comment (onobc) [inline tools/kafka/docker-compose/mysql.yml:1]: These are slated to go in under `tools/kafka`. I wonder if we should instead add a `tools/mysql` and `tools/postgres`? Or maybe this is instead under `tools/scs-schema-registry`. Wdyt?

File: samples/schema-registry-integration/pom.xml [hunk 2]
```diff
@@ -0,0 +1,133 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (onobc) [inline samples/schema-registry-integration/pom.xml:2]: As we already have a `confluent-schema-registry-integration` should we prefix this w/ `scs-` (or some other prefix?).

File: samples/schema-registry-integration/README.adoc [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+Producer V1 (`producer1`), Producer V2 (`producer2`), and Consumer (`consumer`) are included in this project.
+
+=== Requirement
+As a developer, I'd like to design my consumer to be resilient to differing payload schemas.
+
+=== Assumptions
+For this demonstration, we will simply assume there are two producers producing events with different payload schemas.
+A consumer that consumes both the payload versions will be designed to adapt to evolving schemas.
+
+These applications integrate with Spring Cloud Stream Scheam Registry.
```
Comment (onobc) [inline samples/schema-registry-integration/README.adoc:13]: 1. Typo `Scheam`
2. Can be simplified to single sentence
```
Both producers and consumers interact with the SCSt schema registry to register and evolve the schema.
```

File: samples/schema-registry-integration/README.adoc [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+==== Pre-requisites
+****
+* The components have all been built by following the <<build-apps>> steps.
+* Apache Kafka broker available at `localhost:9092`
+
+TIP: The included link:../../../tools/kafka/docker-compose/README.adoc#_all_the_things[Kafka tools] can be used to easily start a broker locally on the required coordinates
+****
+
+If you want to use schema registry with Postgres or MYSQL, either start them locally or use the provided docker-compose scripts above.
```
Comment (onobc) [inline samples/schema-registry-integration/README.adoc:36]: > or use the provided docker-compose scripts above.

So that README does not talk about `postgres` or `mysql`. We should add a section in there for these 2 new additions. Also, this would be influenced by the outcome [here](https://github.com/spring-cloud/spring-cloud-stream/pull/2498/files#r957741938).


I would reword this slightly such as:
```
By default, the schema registry is backed by an H2 database. If you want to instead use Postgres, it must be available at `localhost:5432`. Similarly, if you want to use MySQL it must be available at `localhost:3306`. 

TIP: The included link:../../../tools/kafka/docker-compose/README.adoc#_my_new_section_goes_here[Kafka tools] can be used to easily start a broker locally on the required coordinates.

```

File: samples/schema-registry-integration/README.adoc [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+- Start the Schema Registry server (Change these commands accordingly if you are not on a Unix like platform)
+[source,bash]
+----
+java -jar <ROOT-OF-SPRING-CLOUD-STREAM-CHECKOUT>/schema-registry/spring-cloud-stream-schema-registry-server-<VERSION>.jar
+----
+
+This starts the schema registry with a local H2 database.
+
+To use Postgres database instead of H2, set the following property :
```
Comment (onobc) [inline samples/schema-registry-integration/README.adoc:49]: I would reword slightly to say 
```
To use Postgres instead of H2, additional properties must be specified when starting the server:
```

Same for MySQL instructions below.

File: samples/schema-registry-integration/schema-registry-consumer-kafka/mvnw [hunk 6]
```diff
@@ -0,0 +1,226 @@
+#!/bin/sh
+# ----------------------------------------------------------------------------
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
```
Comment (onobc) [inline samples/schema-registry-integration/schema-registry-consumer-kafka/mvnw:4]: I believe we can pull the wrapper up to the `schema-registry-integration` folder and avoid triplicating this. I did that for the confluent sample.

File: samples/schema-registry-integration/schema-registry-consumer-kafka/src/main/java/sample/consumer/ConsumerApplication.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+	private final Log logger = LogFactory.getLog(getClass());
+
+	public static void main(String[] args) {
+		SpringApplication.run(ConsumerApplication.class, args);
+	}
+
+	@Bean
+	public Consumer<Sensor> process()  {
+			return input -> logger.info("input: " + input);
```
Comment (onobc) [inline samples/schema-registry-integration/schema-registry-consumer-kafka/src/main/java/sample/consumer/ConsumerApplication.java:25]: **[NIT]** Spacing

File: samples/schema-registry-integration/schema-registry-consumer-kafka/src/main/java/sample/consumer/ConsumerApplication.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+import com.example.Sensor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.stream.schema.registry.client.EnableSchemaRegistryClient;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
```
Comment (onobc) [inline samples/schema-registry-integration/schema-registry-consumer-kafka/src/main/java/sample/consumer/ConsumerApplication.java:13]: Do you mind adding `proxyBeanMethods = 'false`?

File: samples/schema-registry-integration/schema-registry-consumer-kafka/src/main/resources/application.yml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+      bindings:
+        process-in-0:
+          destination: sensor-topic
+      schema-registry-client:
+        endpoint: http://localhost:8990
+      schema:
+        avro:
+          schema-locations: classpath:avro/sensor.avsc
+server.port: 9999
+spring.cloud.stream.function.definition: process
```
Comment (onobc) [inline samples/schema-registry-integration/schema-registry-consumer-kafka/src/main/resources/application.yml:13]: Any reason to not merge this in the line-breaked yaml above?

File: samples/schema-registry-integration/schema-registry-producer1-kafka/src/main/java/sample/producer1/Producer1ApplicationKafka.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cloud.stream.function.StreamBridge;
+import org.springframework.cloud.stream.schema.registry.client.EnableSchemaRegistryClient;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+@SpringBootApplication
```
Comment (onobc) [inline samples/schema-registry-integration/schema-registry-producer1-kafka/src/main/java/sample/producer1/Producer1ApplicationKafka.java:17]: **[NIT]** there are a few cleanups we can make (see the [confluent version that I did the same for a while back](https://github.com/spring-cloud/spring-cloud-stream/blob/main/samples/confluent-schema-registry-integration/confluent-schema-registry-integration-producer1/src/main/java/sample/producer1/ConfluentAvroProducer1Application.java)).

**Review by onobc — CHANGES_REQUESTED** [review]
@sobychacko thanks for adding this. Other than a few nits around docs and minor suggestions looks good.

File: tools/kafka/docker-compose/mysql.yml (hunk 1 above)
Comment (sobychacko) [reply tools/kafka/docker-compose/mysql.yml:1]: Good point. I think we can move them to the root of this set of samples (`scst-schema-registry-integration`). They are only applicable to these samples and we don't use them anywhere else.

File: samples/schema-registry-integration/pom.xml (hunk 2 above)
Comment (sobychacko) [reply samples/schema-registry-integration/pom.xml:2]: I am adding `scst-` prefix for all these artifacts.

**Review by onobc — APPROVED** [review]
